### PR TITLE
Alloy ingestion changes

### PR DESCRIPTION
**Pull request by timburke-hackit**
A number of changes to python script:

- Improved handling of api responses
- Remove Pandas dependency
- Writes exported csv to S3 before reading to a spark dataframe (addressing previous failures due to large responses)
- lots of tidying up, removing unnecessary imports, formatting etc

Minor changes to terraform - new required parameter, fixed a trailing space

### CODE REVIEW

File: scripts/jobs/housing_mmh/tenure_restructure.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    glueContext = GlueContext(SparkContext.getOrCreate())
+    logger = glueContext.get_logger()
+    job = Job(glueContext)
+    job.init(args["JOB_NAME"], args)
+
+    # Log something. This will be ouput in the logs of this Glue job [search in the Runs tab: all logs>xxxx_driver]
+    logger.info(f"The job is starting.")
+
+tennure_snapshot = glueContext.create_dynamic_frame.from_catalog(
+    name_space="dataplatform-stg-landing-zone-database",
```
Comment (ryanbratten) [inline scripts/jobs/housing_mmh/tenure_restructure.py:32]: Need environment param

File: scripts/jobs/housing_mmh/tenure_restructure.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    table_name="mtfh_tenureinformation"
+    # pushdown_predicate = create_pushdown_predicate('import_date', 2)
+)
+tennure_snapshot_df = tennure_snapshot.toDF()
+
+tenure_streaming = glueContext.create_dynamic_frame.from_catalog(
+    name_space="dataplatform-stg-landing-zone-database",
```
Comment (ryanbratten) [inline scripts/jobs/housing_mmh/tenure_restructure.py:39]: Need environment param

File: terraform/etl/25-aws-glue-job-env-services.tf [hunk 3]
```diff
@@ -108,7 +108,7 @@ module "alloy_daily_snapshot_env_services" {
     "--increment_prefix"           = "alloy_api_response_"
     "--snapshot_prefix"            = "alloy_snapshot_"
     "--id_col"                     = "item_id"
-    "--increment_date_col "        = "import_datetime"
+    "--increment_date_col"         = "import_datetime"
     "--snapshot_date_col"          = "snapshot_date"
     "--s3_bucket_target"           = "s3://dataplatform-stg-refined-zone/env-services/alloy/snapshots/"
```
Comment (ryanbratten) [inline terraform/etl/25-aws-glue-job-env-services.tf:113]: Need environment param

**Review by ryanbratten — DISMISSED** [review]

File: scripts/jobs/env_services/alloy_api_ingestion.py [hunk 4]
```diff
@@ -1,191 +1,189 @@
...
+        get_last_import_date_time(glue_context, database, glue_catalogue_table_name)
+    )
+
+    logger.info(f"last import date time: {last_import_date_time}")
 
-    if resource == '':
-        raise Exception(
-            "--resource value must be defined in the job aruguments")
+    if resource == "":
+        raise Exception("--resource value must be defined in the job aruguments")
```
Comment (annajgibson) [inline scripts/jobs/env_services/alloy_api_ingestion.py:121]:
```suggestion
        raise Exception("--resource value must be defined in the job arguments")
```

**Review by annajgibson — APPROVED** [review]
Looks good, just one suggestion

**Pull request merged**